File: ios/Splitty/Features/Onboarding/WelcomeGate.swift
import Foundation

/// Когда показывать разовое приветствие.
///
/// Три условия, и каждое взято из живой ситуации:
/// — аккаунт его ещё не видел (ключ по номеру аккаунта, не по устройству);
/// — список групп пуст: у того, кто уже в группах, объяснять нечего, а
///   приветствие поверх работы раздражает;
/// — нет диплинка: человек пришёл по ссылке приглашения в конкретную группу,
///   и показать ему вместо неё рассказ о продукте — значит потерять переход.
func shouldShowWelcome(hasSeen: Bool, groupCount: Int, hasPendingDeeplink: Bool) -> Bool {
    guard !hasSeen else { return false }
    guard groupCount == 0 else { return false }
    return !hasPendingDeeplink
}

File: android/app/src/main/java/com/zagir/splitty/ui/onboarding/WelcomeGate.kt
package com.zagir.splitty.ui.onboarding

/**
 * Когда показывать разовое приветствие.
 *
 * Три условия, и каждое взято из живой ситуации:
 * — аккаунт его ещё не видел (ключ по номеру аккаунта, не по устройству);
 * — список групп пуст: у того, кто уже в группах, объяснять нечего;
 * — нет диплинка: человек пришёл по ссылке приглашения в конкретную группу,
 *   и показать ему вместо неё рассказ о продукте — значит потерять переход.
 */
fun shouldShowWelcome(hasSeen: Boolean, groupCount: Int, hasPendingDeeplink: Boolean): Boolean {
    if (hasSeen) return false
    if (groupCount != 0) return false
    return !hasPendingDeeplink
}
